src/System.Net.HttpListener/src/System/Net/Managed/HttpResponseStream.Managed.cs

```diff
@@ -107,7 +108,30 @@ public override void Close()
             }
         }
 
-        private MemoryStream GetHeaders(bool closing)
+        internal async Task WriteWebSocketHandshakeHeadersAsync()
+        {
+            if (_disposed)
+                throw new ObjectDisposedException(GetType().ToString());
+
+            byte[] bytes = null;
+            MemoryStream ms = GetHeaders(false, true);
+            bool chunked = _response.SendChunked;
+            if (_stream.CanWrite)
```

Comment:
Seems like the three lines above (starting from ```byte[] bytes = null``` should also be inside the if block.